lib/Wikia/src/Domain/User/Preferences/UserPreferences.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Wikia\Domain\User\Preferences;
+
+class UserPreferences {
+
+	/** @var GlobalPreference[] */
+	private $globalPreferences;
+
+	/** @var array[wikiId => [name => LocalPreference]] */
+	private $localPreferences;
+
+	function __construct() {
+		$this->globalPreferences = [];
+		$this->localPreferences = [];
+	}
+
+	public function setGlobalPreference($name, $value) {
+		$this->globalPreferences[$name] = new GlobalPreference($name, $value);
+		return $this;
+	}
+
+	public function setLocalPreference($name, $wikiId, $value) {
+		$this->localPreferences[$wikiId][$name] = new LocalPreference($name, $value, $wikiId);
```

Comment:
I was thinking more to put a constraints on that implementation to have global preferences and local preferences for one wiki, to limit the possible size of local ones, as request to MW use local preferences only for specific wiki.